src/rgw/rgw_rest_s3.cc

```diff
@@ -1268,9 +1268,13 @@ struct ReplicationConfiguration {
       pipe->id = id;
       pipe->params.priority = priority;
 
-      const auto& user_id = s->user->get_id();
+      string tenant_owner;
+      int r = get_owner_tenant(s, null_yield, driver, s->owner.id, tenant_owner);
```

Comment:
does `s->auth.identity->get_tenant()` work here? getting it directly from the identity would avoid the overhead of `driver->load_account_by_id()`

Reply:
No, unfortunately, that is still pointing to the `user_info` which has the system account (forwarded request) info. But we can introduce a function called `get_aclowner_tenant` (not sure if it's a good idea to override the current `get_tenant`) to return the effective_owner's tenant loaded here https://github.com/ceph/ceph/blob/main/src/rgw/rgw_auth_filters.h#L287-L294

Reply:
I have pushed a commit representing the idea of `get_aclowner_tenant`.